### PR TITLE
light-client: Add testing-related documentation

### LINKED ISSUE #595: docs: Document testing of light-client crate
We need comprehensive documentation for the ways in which we're testing the `light-client` crate in its [README](https://github.com/informalsystems/tendermint-rs/blob/master/light-client/README.md).

### PR DESCRIPTION
**Pull request by romac**
📖 [**Rendered**](https://github.com/informalsystems/tendermint-rs/blob/romac/lc-testing-doc/light-client/README.md#testing)

Close #595

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: light-client/README.md [hunk 1]
```diff
@@ -59,6 +59,87 @@ loop {
...
+
+This project also includes simple integration test which spawns a light client instance
+against a single Tendermint full node which acts both as a primary peer and as its
+own witness.
+
+Because this test requires a running Tendermint node, it is ignored by default.
+To run this test locally:
+
+```bash
+$ cargo test -p tendermint-light-client --test integration -- --ignored --nocapture
```
Comment (thanethomson) [inline light-client/README.md:140]: Would it perhaps be a good idea here to be more explicit about the need to run a Tendermint node in order to run these integration tests?

```suggestion
# In one terminal
$ mkdir -p /tmp/tendermint
$ docker run -it --rm -v "/tmp/tendermint:/tendermint" tendermint/tendermint init
$ docker run -it --rm -v "/tmp/tendermint:/tendermint" -p 26657:26657 tendermint/tendermint node --proxy_app=kvstore

# In another terminal
$ cargo test -p tendermint-light-client --test integration -- --ignored --nocapture
```

**Review by thanethomson — COMMENTED** [review]
I see that, in CI, the Light Client's tested by way of:

```
cargo test --all-features --no-fail-fast -p tendermint-light-client
```

Are any additional tests executed in CI but not mentioned here in the `README.md`? (And if so, are they worth mentioning?)

File: light-client/README.md (hunk 1 above)
Comment (romac) [reply light-client/README.md:140]: Good idea!

**Comment by romac** [comment]
Right, there are few unit tests for some data structures (eg. PeerList) here and there as well, which I did not mention. Will fix that and suggest running the command above to run all tests.

**Comment by andrey-kuprianov** [comment]
@romac great overview!

should I add here a few lines about model-based tests? They are also currently executed within the CI, I think.

**Comment by romac** [comment]
@andrey-kuprianov Thanks! 

> should I add here a few lines about model-based tests? 

Yes, please :)

**Review by thanethomson — APPROVED** [review]

**Pull request merged**